R/ordering.R

```diff
@@ -82,3 +82,90 @@ setMethod("ordering<-", "DatasetCatalog", function (x, value) {
     }
     return(x)
 })
+
+#' Move a variable to after another variable
+#'
+#' @param x the variable group that `after` is in
+#' @param value the variable or dataset subset you would like to move
+#' @param after the variable you want to precede `value`
+#' @return returns a variable group
+#' @examples
+#' \dontrun{
+#' ordering(ds)[['Demographics']] <- moveToAfter(ordering(ds)[['Grp A']], ds['age'], ds$educ)
+#' }
+moveToAfter <- function(x, value, after){
+    if (!inherits(after, "OrderGroup")) {
+        after <- urls(after)
+    }
+    if (!inherits(value, "OrderGroup")) {
+        value <- urls(value)
+    }
+    whi <- ifelse(inherits(after, "OrderGroup"), which(names(x) %in% name(after)), which(entities(x) %in% after))
+    if ((inherits(value, "OrderGroup") && !name(value) %in% names(x)) || (!inherits(value, "OrderGroup") && !value %in% entities(x))) entities(x) <- c(entities(x), value)
+    whi2 <- ifelse(inherits(value, "OrderGroup"), which(names(x) %in% name(value)), which(entities(x) %in% value))
+    entities(x) <- entities(x)[c(setdiff(1:whi, whi2), whi2, setdiff((whi+1):length(entities(x)), whi2))]
+    return(x)
+}
+
+#' Copy the order from one dataset to another.
+#'
+#' @param source the dataset you want to copy the order from
+#' @param target the dataset you want to copy the order to
+#' @return returns an object of class [`VariableOrder`] (which can be assigned
+#' to a dataset with [`ordering`])
+#' @examples
+#' \dontrun{
+#' ordering(ds) <- copyOrder(ds1, ds)
+#' }
+#' @export
+copyOrder <- function(source, target){
+    if (!is.dataset(source) | !is.dataset(target)) {
+        halt("Both source and target must be Crunch datasets.")
+    }
+
+    ord <- entities(ordering(source))
+    new_ord <- lapply(ord, copyOrderGroup,  source = source, target = target)
+
+    # TODO: copy other VariableOrder features?
+
+    return(do.call(VariableOrder, new_ord))
+}
+
+#' Copy the order of a `VariableGroup` (or individual variable url) from `VariableOrder`
+#'
+#' @param gr the group or variable url to be copied
+#' @param source the dataset you want to copy the order from
+#' @param target the dataset you want to copy the order to
+#' @return returns either a [`VariableGroup`] (if a group is supplied) or a url (if just a variable url is supplied)
+#' @examples
+#' \dontrun{
+#' ordering(ds) <- copyOrder(ds1, ds)
+#' }
+#' @keywords internal
+copyOrderGroup <- function(gr, source, target) {
```

Comment:
This would be simpler (and probably faster) if instead of passing in datasets and calling this `aliasFromURL` repeatedly, pass in two vectors or lists: `source` has URLs as names and aliases as values, and `target` has aliases as names and URLs as values.